Part1_InterruptControllersEvolution.html

```diff
@@ -34,30 +34,30 @@ <h3>Introduction</h3>
 
 	<p>This article is about hardware/external interrupts IRQ.<p/>
 
-	<p>What is the purpose of interrupts? For example, we want to make some action with an incoming packet from the network card, as soon as the packet arrives. If you don't want to continuously ask the network card "Have my packet arrived?" and waste your processor time, you can make use of external hardware interrupt IRQ. Interrupt line from a device should be connected to INTR line of CPU, and after each packet receive, the network card will make a signal over this line. CPU will sense this signal, and know that network card has information for it. Only after that CPU will read the incoming packet.<p/>
+	<p>What is the purpose of interrupts? For example, we want to perform an action with an incoming packet from the network card as soon as the packet arrives. If you don't want to continuously ask the network card "Has my packet arrived?" and waste your processor time, you can use external hardware interrupt IRQ. The interrupt line from a device should be connected to the INTR line of the CPU, and after each packet is received, the network card will make a signal over this line. The CPU will sense this signal and know that the network card has information for it. Only after that will the CPU read the incoming packet.<p/>

```

Comment:
"Only after that will the CPU read the incoming packet" - is this sentence structure really correct? It sounds strange to me

Reply:
Hmmm, I think it's correct (or at least it sounds natural to me). "Only after that the CPU will read the incoming packet" also works.

Reply:
```suggestion
	<p>What is the purpose of interrupts? For example, we want to perform an action with an incoming packet from the network card as soon as the packet arrives. If you don't want to continuously ask the network card "Has my packet arrived?" and waste your processor time, you can use external hardware interrupt IRQ. The interrupt line from a device should be connected to the INTR line of the CPU, and after each packet is received, the network card will make a signal over this line. The CPU will sense this signal and know that the network card has information for it. Only after that the CPU will read the incoming packet.<p/>
```